app/src/main/java/com/github/se/eduverse/ui/dashboard/DashboardScreen.kt

```diff
@@ -0,0 +1,153 @@
+import com.github.se.eduverse.ui.navigation.NavigationActions
+import com.github.se.eduverse.viewmodel.DashboardViewModel
+import com.google.firebase.auth.FirebaseAuth
+
+var auth: FirebaseAuth = FirebaseAuth.getInstance()
+
+@OptIn(ExperimentalMaterialApi::class)
+@SuppressLint("UnusedMaterialScaffoldPaddingParameter")
+@Composable
+fun DashboardScreen(
+    navigationActions: NavigationActions,
+    viewModel: DashboardViewModel,
+    userId: String = auth.currentUser!!.uid
+) {
+  val widgetList by viewModel.widgetList.collectAsState()
+  var showAddWidgetDialog by remember { mutableStateOf(false) }
+
+  LaunchedEffect(userId) { viewModel.fetchWidgets(userId) }
+
+  var deleteWidgetId by remember { mutableStateOf<String?>(null) }
+
+  Scaffold(
+      floatingActionButton = {
+        FloatingActionButton(
+            onClick = { showAddWidgetDialog = true }, // Show dialog to add widget
+            modifier = Modifier.testTag("add_widget_button")) {
+              Icon(Icons.Default.Add, contentDescription = "Add Widget")
+            }
+      },
+      bottomBar = {
+        BottomNavigationMenu(
+            onTabSelect = { route -> navigationActions.navigateTo(route) },
+            tabList = LIST_TOP_LEVEL_DESTINATION,
+            selectedItem = navigationActions.currentRoute())
+      },
+      modifier = Modifier.testTag("dashboard_screen")) {
+        Box {
+          Column(modifier = Modifier.fillMaxSize()) {
+            LazyColumn(modifier = Modifier.testTag("widget_list")) {
+              items(widgetList) { widget ->
+                WidgetCard(
+                    widget = widget,
+                    showDeleteIcon = deleteWidgetId == widget.widgetId,
+                    onLongPress = {
+                      deleteWidgetId =
+                          if (deleteWidgetId == widget.widgetId) null else widget.widgetId
+                    },
+                    onDeleteClick = {
+                      viewModel.removeWidget(widget.widgetId)
+                      Log.d("DashboardScreen", "Widget removed: ${widget.widgetId}")
+                      viewModel.fetchWidgets(userId)
+                      deleteWidgetId = null
+                    })
+              }
+            }
+          }
+
+          if (showAddWidgetDialog) {
+            CommonWidgetDialog(
+                viewModel = viewModel, onDismiss = { showAddWidgetDialog = false }, userId = userId)
+          }
+        }
+      }
+}
+
+@Composable
+fun CommonWidgetDialog(viewModel: DashboardViewModel, onDismiss: () -> Unit, userId: String) {
+  val commonWidgets = viewModel.getCommonWidgets()
+
+  AlertDialog(
+      onDismissRequest = onDismiss,
+      title = { Text("Add Widget") },
+      text = {
+        Column {
+          commonWidgets.forEach { widget ->
+            Row(
+                modifier =
+                    Modifier.fillMaxWidth()
+                        .padding(vertical = 4.dp)
+                        .clickable {
```

Comment:
Given my comment above, I suggest here applying a for each on the list of widgetOptions which can be a list of widgetType and perform a switch on each element and for each type a call a specific helper function to handle the creation of the widget.